raiden-ts/src/channels/utils.ts

```diff
@@ -165,31 +173,71 @@ export function assertTx(
   { log }: Pick<RaidenEpicDeps, 'log'>,
 ): OperatorFunction<
   ContractTransaction,
-  ContractReceipt & { transactionHash: Hash; blockNumber: number }
+  [ContractTransaction, ContractReceipt & { transactionHash: Hash; blockNumber: number }]
 > {
-  /**
-   * Operator to check for tx
-   *
-   * @param tx - pending contract tx
-   * @returns Observable of txHash
-   */
-  return (tx) =>
-    tx.pipe(
+  return (tx$) =>
+    tx$.pipe(
       tap((tx) => log.debug(`sent ${method} tx "${tx.hash}" to "${tx.to}"`)),
-      mergeMap((tx) => tx.wait()),
-      map((receipt) => {
+      mergeMap(async (tx) => [tx, await tx.wait()] as const),
+      map(([tx, receipt]) => {
         if (!receipt.status || !receipt.transactionHash || !receipt.blockNumber)
           throw new RaidenError(error, {
             status: receipt.status ?? null,
             transactionHash: receipt.transactionHash ?? null,
             blockNumber: receipt.blockNumber ?? null,
           });
         log.debug(`${method} tx "${receipt.transactionHash}" successfuly mined!`);
-        return receipt as ContractReceipt & { transactionHash: Hash; blockNumber: number };
+        return [tx, receipt] as [
+          ContractTransaction,
+          ContractReceipt & { transactionHash: Hash; blockNumber: number },
+        ];
       }),
     );
 }
 
+export const txNonceErrors: readonly string[] = [
```

Comment:
Where did you took these errors from? No chance to get them actually as "code" from EthersJS? :thinking:

Reply:
These varies from eth node being used (parity, geth, infura?). These codes I got in practice, by triggering the bug several times with the scenarios.